src/WebClient.ts

```diff
@@ -26,16 +26,61 @@ const pkg = require('../package.json'); // tslint:disable-line:no-require-import
  * a convenience wrapper for calling the {@link WebClient#apiCall} method using the method name as the first parameter.
  */
 export class WebClient extends EventEmitter {
+
+  /**
+   * Authentication and authorization token for accessing Slack Web API (usually begins with `xoxa`, xoxp`, or `xoxb`)
+   */
+  public get token(): string | undefined {
+    return this._accessToken;
+  }
+  public set token(newToken: string | undefined) {
+    this.accessTokenExpiresAt = undefined;
+    this.isTokenRefreshing = false;
+    this._accessToken = newToken;
+  }
+
   /**
-   * Authentication and authorization token for accessing Slack Web API (usually begins with `xoxp`, `xoxb`, or `xoxa`)
+   * OAuth 2.0 refresh token used to automatically create new access tokens (`token`) when the current is expired.
    */
-  public readonly token?: string;
+  public readonly refreshToken?: string;
+
+  /**
+   * OAuth 2.0 client identifier
+   */
+  public readonly clientId?: string;
+
+  /**
+   * OAuth 2.0 client secret
+   */
+  public readonly clientSecret?: string;
 
   /**
    * The base URL for reaching Slack's Web API. Consider changing this value for testing purposes.
    */
   public readonly slackApiUrl: string;
 
+  /**
+   * The backing store for the current access token.
+   */
+  private _accessToken?: string;
+
+  /**
+   * The time (in milliseconds) when the current access token will expire
+   */
+  private accessTokenExpiresAt?: number;
```

Comment:
Should be `accessTokenExpiresIn`, it's a countdown TTL, not an absolute timestamp of the time of expiration.

Reply:
the response contains `expires_in`, you're right. but the client only stores that value after it adds it to the current time, so this is still correct. it wouldn't be useful to store just the `expires_in` because we would have to compute the future time that the expiration will occur on every request. this way, we only do that computation once and store it. my biggest concern was also clearing this value at the appropriate time, which is handled by the setter.

Reply:
Ah. I missed that. Carry on!